presto-main/src/main/java/io/prestosql/metadata/HandleResolver.java

```diff
@@ -42,26 +42,25 @@
 import static io.prestosql.operator.ExchangeOperator.REMOTE_CONNECTOR_ID;
 import static java.util.Objects.requireNonNull;
 
-public class HandleResolver
+public final class HandleResolver
 {
     private final ConcurrentMap<String, MaterializedHandleResolver> handleResolvers = new ConcurrentHashMap<>();
 
     @Inject
     public HandleResolver()
     {
-        handleResolvers.put(REMOTE_CONNECTOR_ID.toString(), new MaterializedHandleResolver(new RemoteHandleResolver()));
-        handleResolvers.put("$system", new MaterializedHandleResolver(new SystemHandleResolver()));
-        handleResolvers.put("$info_schema", new MaterializedHandleResolver(new InformationSchemaHandleResolver()));
-        handleResolvers.put("$empty", new MaterializedHandleResolver(new EmptySplitHandleResolver()));
+        addConnectorName(REMOTE_CONNECTOR_ID.toString(), new RemoteHandleResolver());
+        addConnectorName("$system", new SystemHandleResolver());
+        addConnectorName("$info_schema", new InformationSchemaHandleResolver());
+        addConnectorName("$empty", new EmptySplitHandleResolver());
     }
 
-    public void addConnectorName(String name, ConnectorHandleResolver resolver)
+    public void addConnectorName(String catalogName, ConnectorHandleResolver resolver)
     {
-        requireNonNull(name, "name is null");
+        requireNonNull(catalogName, "catalogName is null");
         requireNonNull(resolver, "resolver is null");
-        MaterializedHandleResolver existingResolver = handleResolvers.putIfAbsent(name, new MaterializedHandleResolver(resolver));
-        checkState(existingResolver == null || existingResolver.equals(resolver),
-                "Connector '%s' is already assigned to resolver: %s", name, existingResolver);
+        MaterializedHandleResolver existingResolver = handleResolvers.putIfAbsent(catalogName, new MaterializedHandleResolver(resolver));
```

Comment:
It is fine.  The job of the resolver is to map between a `String` id and a `Class` object.  If the classes are the same than any of the ids work, because on the other side of the transport we will convert the id back to a class.  In the case of multiple class loaders in this patch for Hive and Iceberg, the class objects are different (even though they have the same name), so it also works for them.